### ISSUE TITLE
Why wiki.js changes underscores to dashes when files are uploaded?

### ISSUE THREAD

**Issue by Rogach** [Requarks/wiki#304]
### Actual behavior

I upload `a_b.png` - get `a-b.png`.

### Expected behavior

Expected to have `a_b.png` in my file list.

**Comment by Rogach** [comment]
Also, other name mangling happens - `s24` becomes `s-24`. Since I'm uploading lab project files to wiki, I need to keep names clean, and I use combination of various numbers, underscores, and dashes - changing underscores to dashes really screws everything up.

**Comment by NGPixel** [comment]
Because Wiki.js relies on kebab-case filenames to read and write data.

I'll think of a solution for 2.0 to get around this limitation.

**Comment by Rogach** [comment]
Yes, that I understood. But was there any reason for that kebab-case? Manually uploading the files and setting proper names in the links works without any problems.

**Comment by NGPixel** [comment]
Non-technical users are really bad at using proper file naming. They include spaces, accents and other non url friendly characters, which is why all filenames go through a kebab-case conversion first.

**Comment by Rogach** [comment]
Okay, I can understand that non-url-friendly characters are bad, but adding dashes between letters and numbers and replacing underscores seems a little too much.

Also, why force url-friendliness at all? There will always be non-latin languages, which are completely url-unfriendly (all or almost all characters are encoded). Maybe it is easier to just leave all the names as-is?
